src/components/AddItemForm.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { format } from 'prettier';
+import { withFirestore } from 'react-firestore';
+import '../CSS/AddItemForm.css';
+class AddItemForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      next_purchase: null,
+      last_purchase: null,
+      user_token: 'shpongle',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ name: event.target.value });
+  }
+
+  handleSubmit(event) {
+    this.props.firestore
+      .collection('itemform')
+      .add(this.state)
+      .then(function(docRef) {
+        console.log('Document written with ID: ', docRef.id);
+      })
+      .catch(function(error) {
+        console.error('Error adding document:', error);
+      });
+    event.preventDefault();
+    //To reset the input field after the user hits submit
+    this.setState({
+      name: '',
+      ext_purchase: null,
```

Comment:
Tyypo . . . : )

Reply:
You could make the state reset more consistent by defining an `initialState` variable. 

```jsx
const initialState = {
  name: 'some default or blank'
}

class Form extends Component {
  state = initialState

  handleSubmit = event => {
    ...submitLogic
    this.setState(initialState);
  }
}
```